components/camera/client.go

```diff
@@ -228,6 +179,34 @@ func (c *client) Stream(
 	return stream, nil
 }
 
+func (c *client) Image(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, ImageMetadata, error) {
+	ctx, span := trace.StartSpan(ctx, "camera::client::Image")
+	defer span.End()
+	expectedType, _ := utils.CheckLazyMIMEType(mimeType)
+
+	convertedExtra, err := goprotoutils.StructToStructPb(extra)
+	if err != nil {
+		return nil, ImageMetadata{}, err
+	}
+	resp, err := c.client.GetImage(ctx, &pb.GetImageRequest{
+		Name:     c.name,
+		MimeType: expectedType,
+		Extra:    convertedExtra,
+	})
+	if err != nil {
+		return nil, ImageMetadata{}, err
+	}
+
+	if expectedType != "" && resp.MimeType != expectedType {
```

Comment:
A little confused about the handling here.

What happens here if expectedType is empty? Does this mean we fill with the empty mime type to signal we do not know?

Reply:
These are really good questions that I'm gonna take a shot at answering. Full transparency, I think I copied this logic over from the deleted `Read` func so not 100% sure on the original intent

Reply:
Uhh in regards to this particular question, when the expectedType is empty, I think it just skips the log and returns the response mimetype

Reply:
expectedType and mimeType from the argument can just be "", and it's fine. That's a signal to the server to give back whatever bytes it wants. 

Sometimes though people can request a specific mimeType through the client, and then that request will be sent to the server. But sometimes the server doesn't have that mimeType, but still returns an image, encoded with a different type.

so if expectedType == "", it means the client doesn't care what kind of bytes you get back.

but if the client had requested a specific type, and resp.MimeType != expectedType, then it means the server returned a different type than was requested.